src/Data/Aeson/V2/Key.hs

```diff
@@ -0,0 +1,53 @@
+{-# LANGUAGE CPP #-}
+
+module Data.Aeson.V2.Key
+  ( -- * @2.0.0.0@
+    Key
+  , fromString
+  , toString
+  , toText
+  , fromText
+  , coercionToText
+
+    -- * @2.0.2.0@
+  , toShortText
+  , fromShortText
+  ) where
+
+import Prelude
+
+#if MIN_VERSION_aeson(2, 0, 0)
+import Data.Aeson.Key
+#endif
+
+import Data.Text (Text, pack, unpack)
+import Data.Text.Short (ShortText)
+import qualified Data.Text.Short as ST
+import Data.Type.Coercion (Coercion (..))
+
+#if !MIN_VERSION_aeson(2, 0, 0)
+type Key = Text
```

Comment:
```suggestion
newtype Key = Key Text
```

Again in the interest of making sure that changing the `aeson` version doesn't change the API. As a type alias, afaik there's no way of avoiding exposing the fact that it's a type alias, so users will be able to use `Key` and `Text` interchangeably when building with aeson 1 but not when building with aeson 2.

Reply:
I think that kind of breaks the whole idea though... If you're working with `aeson-1.x` it has `type Object = HashMap Text Value` (for example), and `aeson-2.x` has `type Object = KeyMap Key Value`. The only way for one to write code that works against both dependencies if with `type Key = Text` and `type KeyMap = HashMap` defined for the former. Right?

Reply:
> IMO this is necessary, a fundamental aspect of what this project does

I think what you're describing is basically fixing what aeson did wrong by forcing everything (everyone) to go through our package instead, which abstracts it away. It's definitely "morally correct", but I just don't see that ever being successful, so the practical outcome is we just do a lot more work to make things more annoying for ourselves.

If that's the end goal you'd like to see, I'm happy to hand this project off to you and wish you godspeed. But I'm not going to build it. I think, even if it exists, I would much rather just duplicate the `Key = Text` CPP in any projects that need it.

Reply:
Okay, that gives me a better sense of the goal.